Improve wallet and provider integration testing

IAvecilla opened the pull request.

ilitteri reviewed (approved):
Very nice! You could use it his PR or we could use another one but it'd be nice to have a test_helpers module.

IAvecilla commented:
> Very nice! You could use this PR or we could use another one but it'd be nice to have a test_helpers module.

I used this same PR to make these changes, and I modified some places where we were not using the test utils. I also renamed the file to just `utils` since it is inside the `test` directory. Feel free to comment on these changes 🙂

jpcenteno reviewed (approved):
LGTM
